sdk/identity/azure-identity/azure/identity/_internal/__init__.py

```diff
@@ -43,6 +49,24 @@ def validate_tenant_id(tenant_id):
         )
 
 
+def resolve_tenant(default_tenant, allow_multitenant, tenant_id=None, **_):
```

Comment:
@christothes I believe this is equivalent to your `TenantIdResolver` but please take a look, maybe I misread that (C# pattern matching syntax is new to me).